python/tests/algorithms/test_basic.py

```diff
@@ -18,6 +24,37 @@
 from helper import get_live_seismogram, get_live_timeseries, get_sin_timeseries
 
 
+def test_ExtractComponent():
```

Comment:
The tests here is far from sufficient. First, you need to make sure most (if not all) the new codes are covered, and you should double check the coverage [here](https://app.codecov.io/gh/mspass-team/mspass/pull/338). Then, you need to make sure the tests also cover reasonable amount of corner cases. This is especially important to ensure exceptions are captured as expected.